packages/wrangler/src/experimental-flags.ts

```diff
@@ -0,0 +1,16 @@
+import { AsyncLocalStorage } from "async_hooks";
+
+type ExperimentalFlags = {
+	DEV_ENV: boolean;
+	FILE_BASED_REGISTRY: boolean;
+};
+
+const flags = new AsyncLocalStorage<ExperimentalFlags>();
+
+export const run = <V>(flagValues: ExperimentalFlags, cb: () => V) =>
+	flags.run(flagValues, cb);
+
+export const FILE_BASED_REGISTRY = () => flags.getStore()?.FILE_BASED_REGISTRY;
+
+// TODO: actually use this
+export const DEV_ENV = () => flags.getStore()?.DEV_ENV;
```

Comment:
Can we discuss alternatives to this interface please?

Reply:
Minor point about interface:

I'd like to make adding a feature flag as easy as adding it to a type with no repetition. Having an exported function per flag requires multiple changes to this file to add each flag. While this allows us to find all usages of _each_ feature flag with Find All References, they won't be easy to find usages of _any_ flag. Can we try something like `export const XFLAG = <F extends keyof ExperimentalFlags>(flag: F) => ExperimentalFlags[F];`?

More for discussion:

This is a great use of AsyncLocalStorage. Can we generalise this use for other singletons like Logger etc? Are there any parts of wrangler this wouldn't work for?